src/main/java/com/rakovets/course/java/core/practice/oop_classes_and_objects/Atm.java

```diff
@@ -0,0 +1,111 @@
+        int totalAmountInAtm = numberBanknotes20 * VALUE_BANKNOTES_20 + numberBanknotes50 * VALUE_BANKNOTES_50 + numberBanknotes100 * VALUE_BANKNOTES_100;
+        if (amount > 0 && amount < totalAmountInAtm) {
+            for (int i = 0; i <= (amount / VALUE_BANKNOTES_100) && i <= numberBanknotes100; i++) {
+                for (int j = 0; j <= (amount / VALUE_BANKNOTES_50) && j <= numberBanknotes50; j++) {
+                    for (int k = 0; k <= (amount / VALUE_BANKNOTES_20) && k <= numberBanknotes20; k++) {
+                        if (amount == k * VALUE_BANKNOTES_20 + j * VALUE_BANKNOTES_50 + i * VALUE_BANKNOTES_100) {
+                            isPossibleIssue = true;
+                            break;
+                        }
+                    }
+                }
+            }
+        } else if (amount == totalAmountInAtm) {
+            isPossibleIssue = true;
+        }
+        if (isPossibleIssue) {
+            currentIssueAmount = amount;
+        }
+        return isPossibleIssue;
+    }
+
+    public int[][] getOptionsCombinationBanknotes(int amount) {
+        int totalNumberOfOptions = 0;
+        int[][] optionsCombinationBanknotes;
+        /* it's much better to use dynamic list of arrays in this case, but we
+        haven't learned this type of objects yet and therefor I use static 2D array */
+        for (int i = 0; i <= amount / VALUE_BANKNOTES_100 && i <= numberBanknotes100; i++) {
+            for (int j = 0; j <= amount / VALUE_BANKNOTES_50 && j <= numberBanknotes50; j++) {
+                for (int k = 0; k <= amount / VALUE_BANKNOTES_20 && k <= numberBanknotes20; k++) {
+                    if (amount == k * VALUE_BANKNOTES_20 + j * VALUE_BANKNOTES_50 + i * VALUE_BANKNOTES_100) {
+                            totalNumberOfOptions += 1;
+                    }
+                }
+            }
+        }
+        if (totalNumberOfOptions > 0) {
+            optionsCombinationBanknotes = new int[totalNumberOfOptions][3];
+            int numberOfOptions = 0;
+            for (int i = 0; i <= amount / VALUE_BANKNOTES_100 && i <= numberBanknotes100; i++) {
+                for (int j = 0; j <= amount / VALUE_BANKNOTES_50 && j <= numberBanknotes50; j++) {
+                    for (int k = 0; k <= amount / VALUE_BANKNOTES_20 && k <= numberBanknotes20; k++) {
+                        if (amount == k * VALUE_BANKNOTES_20 + j * VALUE_BANKNOTES_50 + i * VALUE_BANKNOTES_100) {
+                            optionsCombinationBanknotes[numberOfOptions][0] = k;
+                            optionsCombinationBanknotes[numberOfOptions][1] = j;
+                            optionsCombinationBanknotes[numberOfOptions][2] = i;
+                            numberOfOptions++;
+                        }
+                    }
+                }
+            }
+        } else {
+            optionsCombinationBanknotes = new int[1][3];
+        }
+        return optionsCombinationBanknotes;
```

Comment:
Можно проще

Reply:
not fixed. I understand all disadvatages of this code but kindly ask You to accept this version because of lack of time to prepare next tasks. Thank You!